cin_validator/rules/cin2022_23/rule_4017.py

```diff
@@ -0,0 +1,325 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import CINTable, RuleContext, rule_definition
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+
+ChildProtectionPlans = CINTable.ChildProtectionPlans
+LAchildID = ChildProtectionPlans.LAchildID
+CPPID = ChildProtectionPlans.CPPID
+CPPstartDate = ChildProtectionPlans.CPPstartDate
+CPPendDate = ChildProtectionPlans.CPPendDate
+
+CINplanDates = CINTable.CINplanDates
+CINPlanStartDate = CINplanDates.CINPlanStartDate
+CINPlanEndDate = CINplanDates.CINPlanEndDate
+
+Header = CINTable.Header
+ReferenceDate = Header.ReferenceDate
+
+
+# define characteristics of rule
+@rule_definition(
+    # write the rule code here
+    code=4017,
+    # replace ChildProtectionPlans with the value in the module column of the excel sheet corresponding to this rule .
+    # Note that even if multiple tables are involved, one table will be named in the module column.
+    module=CINTable.ChildProtectionPlans,
```

Comment:
Would you not detail the CIN Plan Dates module here rather than Child Protection Plans?

Reply:
There are two rules that share the same description and look at the same data from opposite angles.  For this one (4017 #86) I've selected Child Protection Plans as the module because the rule is about CPP start date within a CIN Plan.  When I wrote the other side before (4016 #85, PR #207) I selected CIN Plan Dates as the module because that rule is about CIN start date within a CPP.

*I have just noticed though that I forgot to update the rule logic at line 70 here when I copied this from my 4016 branch; will correct that later!